### ISSUE TITLE
Random install errors on Windows CI tests

### ISSUE THREAD

**Issue by JWCook** [requests-cache/requests-cache#797]
I've encountered some weird dependency installation errors in the extended test matrix that runs before deployments, but only on Windows, and only in GitHub Actions: https://github.com/requests-cache/requests-cache/actions/runs/4309110823

Example error:
```
  ChefBuildError

  Backend 'setuptools.build_meta:__legacy__' is not available.

  at ~\.local\venv\lib\site-packages\poetry\installation\chef.py:152 in _prepare
      148| 
      149|                 error = ChefBuildError("\n\n".join(message_parts))
      150| 
      151|             if error is not None:
    > 152|                 raise error from None
      153| 
      154|             return path
      155| 
      156|     def _prepare_sdist(self, archive: Path, destination: Path | None = None) -> Path:

Note: This error originates from the build backend, and is likely not a problem with poetry but with timeout-decorator (0.5.0) not supporting PEP 517 builds. You can verify this by running 'pip wheel --use-pep517 "timeout-decorator (==0.5.0)"'.
```

Sometimes this shows up with `pytest-clarity` instead of `timeout-decorator`, and the test command in the error message (`pip wheel --use-pep517 "timeout-decorator==0.5"`) works fine. The worst part is that it seems to be random; retrying the same job can give different results. So far I can't reproduce it locally.

The test matrix includes combinations of python versions and `requests` versions, but doesn't seem to consistently happen depending on either of those. I'm unsure if this is a problem with those particular libraries, or poetry, pip, setuptools, the build image, or something else I haven't thought of.

**Comment by JWCook** [comment]
I am unsure why, but this problem seems to have resolved itself.

**Comment by JWCook** [comment]
Started seeing this again here: https://github.com/requests-cache/requests-cache/actions/runs/6917875425

It appears that this has been fixed in poetry 1.7 (explanation here: https://github.com/python-poetry/poetry/issues/7611#issuecomment-1747836233). For older versions, the workaround mentioned in that comment works:
```
poetry config installer.max-workers 1
```
